tableau_utilities/scripts/datasource.py

```diff
@@ -1,99 +1,63 @@
-import os
-
+import tableau_utilities.tableau_file.tableau_file_objects as tfo
 from tableau_utilities.tableau_file.tableau_file import Datasource
-from tableau_utilities.tableau_file.tableau_file import TableauFile
-from tableau_utilities.tableau_file.tableau_file_objects import Column
-from tableau_utilities.scripts.server_operate import fill_in_id_name_project, get_object_list
+from tableau_utilities.tableau_server.tableau_server import TableauServer
 
 
-def save_datasource(server, id, datasource_name, project_name, include_extract, debugging_logs):
-    """ Gets the id if needed and downloads the datasource
+def datasource(args, server=None):
+    """ Updates a Tableau Datasource locally
 
     Args:
-        id: The datasource id
-        datasource_name: The name of the datasource
-        project_name: The name of the project the datasource is in
-        include_extract: True to include the extract. Downloading may take a long time
-        debugging_logs: Print information for debugging to the console
-
-
-    """
-
-    object_list = get_object_list(object_type='datasource', server=server)
-    id, datasource_name, project_name = fill_in_id_name_project(id, datasource_name,
-                                                                           project_name, object_list)
-    if debugging_logs:
-        print(
-            f'GETTING DATASOURCE ID: {id}, NAME: {datasource_name}, PROJECT NAME: {project_name}, INCLUDE EXTRACT {include_extract}')
-
-    datasource_path = server.download_datasource(id, include_extract)
-
-    if debugging_logs:
-        print(
-            f'DATASOURCE PATH: {datasource_path}')
-
-    return datasource_path
-
-
-def save_tds_file(datasource_path, extract_to_directory):
-    """ Extracts XML from a tdsx file and saves as a raw .xml file
-
+        args: An argparse args object
+        server (TableauServer): A TableauServer object; optional
     """
-
-    # xml_path = TableauFile(datasource_path).save(save_raw_xml=True)
-    xml_path = TableauFile(datasource_path).unzip(extract_to=extract_to_directory)
-    print(f'TDS SAVED TO: {xml_path}')
-
-
-def datasource(args, server=None):
-
     debugging_logs = args.debugging_logs
-
-    location = args.location
     datasource_path = args.file_path
-    id = args.id
     name = args.name
-    project_name = args.project_name
```

Comment:
@JustinGrilli  I intentionally had the design pattern of putting the args at the top of the function as that helped me know which args were potentially used later on.

Having all the `args.somename` lower in the code as you've changed it to can make it harder to debug. 

Do you mind keeping this design pattern as much as possible?

Reply:
In my opinion, if you are having to dig through the CLI code as the end user, the solution should not be to add redundant variables to the top of that code, it should be to add better debugging/error messages.

Reply:
RE:
> I agree that end users shouldn’t have to dig through the code. I meant that made it easier for me!
Ahh okay, I gotcha. 😅 
Okay, I will try to leave these variables be as much as possible 👍

Reply:
Thanks!